library/Fission/CLI/Command/Watch.hs

```diff
@@ -103,13 +105,38 @@ watcher runner Watch.Options {..} = do
       IPFS.addDir ignoredFiles absPath >>= putErrOr \cid@(CID hash) -> do
         UTF8.putText $ "👀 Watching " <> Text.pack absPath <> " for changes...\n"
 
-        sendRequestM (updateApp url cid copyFiles) >>= putErrOr \_ -> do
+        retrier (updateApp url cid copyFiles) >>= putErrOr \_ -> do
           liftIO $ FS.withManager \watchMgr -> do
             hashCache <- newMVar hash
             timeCache <- newMVar =<< getCurrentTime
             void $ handleTreeChanges runner url copyFiles timeCache hashCache watchMgr absPath
             forever . liftIO $ threadDelay 1000000 -- Sleep main thread
 
+retrier ::
+  ( MonadWebClient m
+  , MonadLogger m
+  )
+  => m (ClientM a)
+  -> m (Either ClientError a)
+retrier req =
+  sendRequestM req >>= \case
+    Right val ->
+      return $ Right val
+
+    Left err@(FailureResponse _req res) -> do
+      let code = statusCode $ responseStatusCode res
+ 
+      if code >= 502 && code <= 504
+        then do
+          logWarn $ "Got a " <> textDisplay code <> "; retrying..."
+          retrier req
+
+        else
+          return $ Left err
+
+    Left err ->
+      return $ Left err
```

Comment:
I believe that this is duplicated — could probably abstract it out, with an extra parameter for the predicate. I can see us retrying other things based on some criteria (in this case the status code).

But also maybe doesn't have to happen _right now_